core/src/test/java/com/ibm/wala/core/tests/cha/AnalysisScopeTest.java

```diff
@@ -55,4 +62,75 @@ public void testBaseScope() throws IOException, ClassHierarchyException {
                 ClassLoaderReference.Application, "Ljava/awt/AlphaComposite")),
         "found unexpected class");
   }
+
+  @Test
+    public void testToJson() throws IOException {
+        AnalysisScope scope =
+            AnalysisScopeReader.instance.readJavaScope(
+                TestConstants.WALA_TESTDATA,
+                // new FileProvider().getFile("J2SEClassHierarchyExclusions.txt"),
+                new FileProvider().getFile("GUIExclusions.txt"),
+                AnalysisScopeTest.class.getClassLoader());
+        Gson gson = new Gson();
+        Type type = new TypeToken<LinkedHashMap<String, Object>>(){}.getType();
+        LinkedHashMap<String, Object> map = gson.fromJson(scope.toJson(), type);
+        System.out.println(map);
+        if(map.containsKey("Exclusions")) {
+            String[] exclusions = scope.getExclusions().toString().split("\\|");
+            ArrayList<String> arr2 = new ArrayList<>();
+            for(int i = 0; i < exclusions.length; i++){
+                String word = exclusions[i];
+                word = word.replace("(", "");
+                word = word.replace(")", "");
+                arr2.add(word);
+            }
+            assertEquals(arr2, map.get("Exclusions"));
+        }
+        Type type2 = new TypeToken<LinkedHashMap<String, ArrayList<String>>>(){}.getType();
+        LinkedHashMap<String, ArrayList<String>> loaders = gson.fromJson(gson.toJson(map.get("Loaders")), type2);
+        if(loaders.containsKey("Primordial")) {
```

Comment:
Again we can simplify this code.  We know the contents of `wala.testdata.txt`:
```
Primordial,Java,stdlib,base
Primordial,Java,jarFile,primordial.jar.model
Application,Java,jarFile,com.ibm.wala.core.testdata_1.0.0.jar
```
So I think we can write assertions like:
* The keys of `loaders` should be exactly `"Primordial","Application","Extension","Synthetic"`
* Extension and Synthetic should be mapped to empty lists
* Primordial should be mapped to a list of length two.  And one of the elements should contain `"primordial.jar.model"
* Application should be mapped to a list of length 1, and that element should contain `"com.ibm.wala.core.testdata_1.0.0.jar"`